source/resampling_with_code2.Rmd

```diff
@@ -610,26 +654,24 @@ z
 
 Finally, we need to count how many of the trials in `z` ended up with all-White juries.  These are the trials with a `z` value of 0.
 
-We can ask this question of *all 100 counts* by asking the question: is the {{< var array_name >}} `z` equal to 0, like this:
+We can ask this question of *all 100 counts* by asking the question: is the {{< var array >}} `z` equal to 0, like this:
 
 ```{python, opts.label="py_ed"}
 # Is the value of z equal to 0?
-all_white = z == 0
+all_white = (z == 0)
 # Show the result of the comparison.
 all_white
 ```
 
 ```{r, opts.label="r_ed"}
 # Is the value of z equal to 0?
-all_white <- z == 0
+all_white <- (z == 0)
```

Comment:
Why the parens?

Reply:
For newcomers, we've found that masks are easier to understand this way.  Otherwise there can be confusion between operator precedence.  They may read it as: `(all_white <- z) == 0`, which makes no sense.

Reply:
Who is "we" there?  Scikit-image?

If 'twere me - I'd be asking - why the brackets?   Do they do something?   What would happen if there were no brackets there?

I think the reader won't be very good yet at thinking what the brackets do, and will be tempted to think that they are some sort of magic here, and so introduce confusion.

Reply:
Readers have brackets from mathematics at early school level, so I don't think it will confuse them nearly as much as the lack of grouping.

Reply:
Well - but here that's pretty opaque.  For example, they won't be used to 

a = (b + c)

That would look very odd to them - and they would surely wonder - "Why the brackets there?  They don't do anything".  And they already know that parentheses mean calling a function.  So my guess would be pretty hard to the other direction (more confusion than clarity).

Reply:
I think you're wrong, but I am going to change it anyway.

Reply:
```
all_white = z == 0
```

vs

```
all_white = (z == 0)
```

In my mind the second clearly reads better, and that is also the style we adopt inside most libraries AFAIK.  It is also the recommended style by Robert Kern :)

Reply:
Done.

Reply:
I think I haven't come across the recommendation - there's probably some source for it somewhere?   And I certainly haven't used it up til now, but I'm sure you've guess that from my question.  But in any case- I think it's a different situation if you know your way around a language.  In that case:

a = (b + c)

won't trip you up, because you're used to reading through brackets.  Although, you saw, I wondered what they were for, and it's hard for me to imagine that beginner readers wouldn't be the same.  I understand the recommendation, and with the explanation, it makes sense, but I think we probably don't want to add an explanation here, and without it, I still think it's just going to be confusing.

Reply:
Expanding - I think any careful student would ask "Why the brackets?".  And I think we don't want to explain at this point.

Reply:
As I mentioned, I think the confusion will be around operator precedence; but I've made the change, so let's continue.

Reply:
Sure - I understood the precedence point.

I think we can deal with that fairly easily with examples like:

```
>>> one_z = 3
>>> one_z == 0
False
>>> two_z = np.array([3, 0])
>>> two_z == 0
array([False,  True])
>>> res = two_z == 0
>>> res
array([False,  True])
```

Reply:
I think simply showing the result of `np.array([...]) == 3` is sufficient.  A concern I have more widely is that very verbose code is harder to parse and learn from.  Sometimes, better to rely on the students inspecting an output and developing intuition from the result.

Reply:
Is it possible that this is a difference between the students I have taught and the ones you have taught?  My students are very easily spooked, when they see something they don't understand, and they have very little confidence in their ability to parse code.  That's where my instinct comes from, to really go the long way round at first, in order to pick up speed later.

Reply:
I guess I use myself as a test case first. I get confused by volume more easily than I do by syntax. But you may well be right that we are used to different audiences.